lib/html-tags.js

```diff
@@ -25,12 +25,14 @@ const voidTags = ['area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'k
  *  A tag element according to the htmlWebpackPlugin object notation
  *
  * @param xhtml {boolean}
- *   Wether the generated html should add closing slashes to be xhtml compliant
+ *   Whether the generated html should add closing slashes to be xhtml compliant
  */
 function htmlTagObjectToString (tagDefinition, xhtml) {
   const attributes = Object.keys(tagDefinition.attributes || {})
     .filter(function (attributeName) {
-      return tagDefinition.attributes[attributeName] !== false;
+      return tagDefinition.attributes[attributeName] !== false &&
+        tagDefinition.attributes[attributeName] !== null &&
+        tagDefinition.attributes[attributeName] !== undefined;
```

Comment:
Ideally, this assertion could be rewritten as the following:

```js
[null, undefined, false].includes(tagDefinition.attributes[attributeName])
```

It has better readability, but due to lint constraints that was not possible.

Reply:
What about `tagDefinition.attributes[attributeName] === '' || tagDefinition.attributes[attributeName]`?

Reply:
This would introduce a bug in the plugin due to the nature of `truthy` values, which excludes `false`, `0`, `""`, `null`, `undefined` and `NaN` values.

As you can see, `0` as value might be needed.

Reply:
but in your typings you didn't allow numbers - should we add them too?

Reply:
that's actually a good point! I'm not sure about the broad use case of it, but if we're going to proceed in this way, then we can simply refactor it to:

```js
  const attributes = Object.keys(tagDefinition.attributes || {})
    .filter(function (attributeName) {
      return tagDefinition.attributes[attributeName];
    })
```

This would satisfy my use case, although I would appreciate your feedback regarding it, as you might have a better sense of the community usage.

Reply:
this would not allow an empty string like: `<input value="">` I am not sure if that's used but removing support for it would be a breaking change...

what about adding a small util function in this file so we can do sth like:

```js
  const attributes = Object.keys(tagDefinition.attributes || {})
    .filter(function (attributeName) {
      return hasValue(tagDefinition.attributes[attributeName]);
    })
```

Reply:
e.g.:

```js
/**
 * @param {undefined | null | string | number | boolean} value
 */
function hasValue(value) {
  if (value === undefined || value === null || value === false) {
    return false;
 }
 return true;
}
```

Reply:
Or we just keep your initial proposal :)

Reply:
The idea to have this `hasValue` is interesting, but if the usage is restricted to this single case, have the assessment inline provides better understand about the code, so I would go for that suggestion of yours:

```js
tagDefinition.attributes[attributeName] === '' || tagDefinition.attributes[attributeName]
```

This fits nicely

Reply:
Okay cool so we have two valid options I guess:

1. The one you already committed (which also allows numbers)
2. The `tagDefinition.attributes[attributeName] === '' || tagDefinition.attributes[attributeName]` which does not allow numbers

I am fine with both ways :)

Reply:
I have applied the suggestion you made, as you have a better understanding about its usage.

Thank you for the nice brainstorming